### ISSUE TITLE
Reduce Number of Tests, Make All Mandatory

### ISSUE THREAD

**Issue by johaq** [RoboCupAtHome/RuleBook#767]
I think back in 2018 we decided we wanted to make the competition less stressful for teams.
That initially meant reducing the number of tests and therefore the number of test slots.
This idea evolved into just reducing the number of test slots but letting teams choose which tests to perform.
This meant instead of reducing the number of tests we actually drastically increased the number of tests so
teams had a choice. 

**I think reversing this descision is the most important change we need to make for next year!**

Letting teams choose the test introduced several problems to the competition:
* Teams' scores are not at all comparable anymore
    * This goes for individual test scores, i.e. teams got 1000 pts for taking pictures of people and placing them on a map, but another got 650 points for picking and placing 4 objects and pouring milk and cereal into a bowl. As well as for the overall scores.
* The league is even more diffictult to follow for the audience
     * Why does one robot store objects into a cabin and the next robot talks to some people?
* Running a testslot becomes a nightmare for referees
    * Everytime a team wants to do storing groceries I need to redecorate the arena? Not to mention that switching constantly between tests just drastically increases the chance of the referee making a mistake. We already have a problem of not having enough people to referee but now we basically created a scenario where a first year referee has to run an incredibly difficult test slot, whereas in the past they could start out by running a Person Recognition test slot for example.

I propose switching back to the old system where every test had its own test slot. To keep in mind the earlier idea of reducing the amount of stress for teams, I propose to reduce the number of tests to:
Stage 1:
* Storing Groceries
* Carry my Luggage
* Receptionist
* Set the Table or Take out the Garbage
* GPSR

Stage 2:
* Restaurant
* Clean the Table
* Smoothie Chef
* EGPSR

I think this gives the easiest to understand tests for the audience while still covering a wide variety of robot skills.
This also gives us back the ability to award points depending on how difficult something is and not based on trying to reach the set amount of points of a test.

**Comment by moriarty** [comment]
I agree.

**Comment by MatthijsBurgh** [comment]
I agree with the idea. I am not sure yet about which challenges.

This of course starts with the focus of the League. Then determine skills to test in the challenges. Then select the challenges from current set or create new ones. But I let this to @PetervDooren.

**Comment by PetervDooren** [comment]
Another aspect that is improved with this is that teams will not pass on difficult but interesting challenges such as the GPSR.

**Comment by MatthijsBurgh** [comment]
Try to find a good balance between keeping the league accessible for new teams and interesting and challenge for established teams

**Comment by ARTenshi** [comment]
I agree. Having several tasks make them redundant; for example, cleaning up can be Clean Up itself where the robot takes objects to their correct place, or Storing Groceries, where robot takes objects to their correct place, or Set and Clean the table, where robots takes objects to their correct place. A team with object detection but without object recognition can still partially score in all those tasks and overcome a team that aims to completely solve one single task. So, a high cumulative score not necessarily represent the best performance. We should find single representative tasks for skill groups and encourage teams to solve it completely to score high.

**Comment by LoyVanBeek** [comment]
Ah, here we get to another contradictory item in scoring for @Home:
Scoring perfectly on subtasks (eg. only perfect object recognition, but no grasping, or vice versa: perfectly grabbing the wrong things) but not achieving the overall goal, how does that score?
A robot not achieving it's goal is still quite useless, but it is discouraging to teams to give 0 points. As soon as you start giving points for subtasks (aka partial scoring), teams generally seem to stop trying for the overall goal of a task.
We used to have some small multiplier (eg. 2x) that was applied when the overall goal was achieved, but this was not enough incentive to go beyond optimizing for subtasks. Maybe there should be a bigger multiplier for achieving the goal, that seems the most obvious fix. 

That being said: since for a task to be completed, even a small hiccup can ruin the whole thing, which is also discouraging. Especially for new teams ,that do not yet have all skills mastered to a sufficient level.
For example, speech recognition in a noisy RoboCup@Home venue is (was, in my time at least) a frequent roadblock, that was often outside the teams and even the TCs control (eg. some soccer league starts whistling, rescue league runs a circular saw, announcer starts blaring).

**Comment by kyordhel** [comment]
Once @LoyVanBeek tried to make scoring consistent among tests and it became a nightmare. Already tried, didn't work. Here I want to advice TC to read the existing papers made by former Trustees and Execs. Focus on scoring subgoals so teams aim to complete the task, and not on point farming.

**Comment by johaq** [comment]
I don't care for the scores being consistent between tests if every test is mandatory. 
It only becomes a problem when teams get to chose the tests.

**Comment by PetervDooren** [comment]
I agree with @johaq, reaching consistency between tasks is impossible. Therefore the best alternative is to make the tests mandatory. This way attempting the difficult task is always preferable to not choosing it. 

I also think that excessively scoring subgoals is what leads to point farming.
